src/librustc/ty/mod.rs

```diff
@@ -755,6 +755,9 @@ pub enum Predicate<'tcx> {
     /// for some substitutions `...` and T being a closure type.
     /// Satisfied (or refuted) once we know the closure's kind.
     ClosureKind(DefId, ClosureKind),
+
+    /// `T1 <: T2`
+    Subtype(PolySubtypePredicate<'tcx>),
```

Comment:
Hmm, perhaps not. I was thinking that in the future a case like `for<'a> fn(?T) <: for<'a> fn(?U)` might cause such a predicate, but that's not really true (the `?T` and `?U` are bound outside the `for`). It would be possible to have higher-ranked *subregion* constraints, in contrast, but those are a separate mechanism.